packages/assets-controllers/src/TokensController.ts

```diff
@@ -624,12 +624,14 @@ export class TokensController extends BaseController<
       });
 
     this.update((state) => {
-      state.ignoredTokens = newIgnoredTokens;
-      state.tokens = newTokens;
       state.detectedTokens = newDetectedTokens;
       state.allIgnoredTokens = newAllIgnoredTokens;
       state.allDetectedTokens = newAllDetectedTokens;
       state.allTokens = newAllTokens;
+      if (interactingChainId === this.#chainId) {
```

Comment:
Is it worth adding an explicit test for this logic? In my opinion the existing test that we have ("should ignore tokens by networkClientId") is very broad and it is unclear based on the name of the test what exactly is being tested.

Reply:
good idea, let me add another explicit test for this logic

Reply:
unit test added [a655a00](https://github.com/MetaMask/core/pull/5014/commits/a655a00d9bb2afe10cebe581d758383f647bcc68)